#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.07482116215151731 0.9893608856497019 0.12476710961120215 4.164416803138813
  position -4.160570028797488 1.3791975552864992 -1.2294353018864927
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 7 7
}
Spot {
  translation -1.28 0.930935 1.36275
  rotation 1 0 0 -0.261795307179586
  controller "SpotOverCart"
}
WoodenBox {
  translation -0.320186 0.05 -0.299581
  rotation 0 1 0 4.692820414042842e-06
  size 0.1 0.1 0.1
}
WoodenBox {
  translation 0.37013 0.05 -0.244191
  rotation 0 1 0 0.523603
  name "wooden box(1)"
  size 0.1 0.1 0.1
}
WoodenBox {
  translation 0.247844 0.05 0.379704
  rotation 0 1 0 -0.785395307179586
  name "wooden box(2)"
  size 0.1 0.1 0.1
}
DEF WALL1 Solid {
  translation 0.505 0.03 0
  children [
    DEF WALL_OBJ Shape {
      appearance PBRAppearance {
        baseColor 0.3 0.3 1
        roughness 0.5
        metalness 0
      }
      geometry Box {
        size 0.01 0.06 1
      }
    }
  ]
  name "solid(1)"
  boundingObject USE WALL_OBJ
}
DEF WALL2 Solid {
  translation -0.505 0.03 0
  children [
    USE WALL_OBJ
  ]
  name "solid(4)"
  boundingObject USE WALL_OBJ
}
DEF WALL3 Solid {
  translation 0 0.03 -0.505
  rotation 0 1 0 1.5707996938995747
  children [
    USE WALL_OBJ
  ]
  name "solid(3)"
  boundingObject USE WALL_OBJ
}
DEF WALL4 Solid {
  translation 0 0.03 0.505
  rotation 0 1 0 1.5707996938995747
  children [
    USE WALL_OBJ
  ]
  name "solid(2)"
  boundingObject USE WALL_OBJ
}
Robot {
  translation 0 0.19 0
  children [
    DistanceSensor {
      translation 0.02 0 0.1
      rotation 0 1 0 -1.0472003061004251
      children [
        DEF SENSOR_BOX Shape {
          appearance PBRAppearance {
            baseColor 0.2 0.3 1
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "dist_sensor_left"
      boundingObject USE SENSOR_BOX
    }
    DistanceSensor {
      translation -0.02 0 0.1
      rotation 0 1 0 -2.094399693899575
      children [
        DEF SENSOR_BOX Shape {
          appearance PBRAppearance {
            baseColor 0.2 0.3 1
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "dist_sensor_right"
      boundingObject USE SENSOR_BOX
    }
    DEF BODY_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.9 0.3 0.2
        roughness 0.2
        metalness 0.4
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF W_FRONT_LEFT HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel_front_left"
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation 0.06000000179392618 -7.889619477799426e-07 0.05000000218490906
        rotation -6.427812726565247e-17 6.715813858658567e-17 1 1.570799693899574
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColorMap ImageTexture {
                url [
                  "textures/wood.jpg"
                ]
              }
              metalness 0
            }
            geometry Cylinder {
              height 0.018
              radius 0.04
            }
          }
        ]
        boundingObject USE WHEEL
        physics DEF WHEEL_PHYSICS Physics {
        }
      }
    }
    DEF W_FRONT_RIGHT HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel_front_right"
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation -0.06 0 0.05
        rotation 0 0 1 1.570799693899574
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColorMap ImageTexture {
                url [
                  "textures/wood.jpg"
                ]
              }
              metalness 0
            }
            geometry Cylinder {
              height 0.018
              radius 0.04
            }
          }
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics DEF WHEEL_PHYSICS Physics {
        }
      }
    }
    DEF W_REAR_LEFT HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel_rear_left"
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation 0.06 0 -0.05
        rotation 0 0 1 1.570799693899574
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColorMap ImageTexture {
                url [
                  "textures/wood.jpg"
                ]
              }
              metalness 0
            }
            geometry Cylinder {
              height 0.018
              radius 0.04
            }
          }
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics DEF WHEEL_PHYSICS Physics {
        }
      }
    }
    DEF W_REAR_RIGHT HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel_rear_right"
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation -0.06 0 -0.05
        rotation 0 0 1 1.570799693899574
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColorMap ImageTexture {
                url [
                  "textures/wood.jpg"
                ]
              }
              metalness 0
            }
            geometry Cylinder {
              height 0.018
              radius 0.04
            }
          }
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics DEF WHEEL_PHYSICS Physics {
        }
      }
    }
  ]
  boundingObject USE BODY_SHAPE
  physics Physics {
  }
  controller "FourWheelerController"
}
PlatformCart {
  translation -2.06 0 1.76
}
